Kconfig.zephyr

```diff
@@ -486,6 +487,11 @@ config BUILD_OUTPUT_UF2_FAMILY_ID
 	  name string. If the SoC in use is known by UF2, the Family ID will
 	  be pre-filled with the known value.
 
+config BUILD_OUTPUT_UF2_OFFSET
+	hex "UF2 Offset"
+	default 0x10000000 if SOC_SERIES_RP2XXX
+	default FLASH_LOAD_OFFSET if HAS_FLASH_LOAD_OFFSET
```

Comment:
This is confusing too. Why not keep this as CONFIG_FLASH_LOAD_OFFSET but set an SoC default for that option in the appropriate SoC file, instead of adding some SoC-specific defaults here in the main Kconfig.zephyr file?

Reply:
Happy to do that, if we want. For the uf2 family, we'd kept it centralized, but I can move this to the SoC if we want.

Reply:
@petejohanson @mbolivar-nordic If I understand this correctly, in most implementations of UF2 the offset is relative to the flash. With the pico, it's absolute (probably because they can use UF2 to program an app straight to RAM and execute it there). This means that we can't assume `CONFIG_FLASH_LOAD_OFFSET` is the correct address to use for creating the UF2. It seems like there should be some kind of logic performed to obtain this value, beyond the capabilities of dts/kconfig. In this instance, it's `0x10000000 if has_bootloader else 0x10000000 + CONFIG_FLASH_LOAD_OFFSET`. I'm not sure how to implement this. Maybe the UF2 offset should be a cmake variable calculated for each SoC? Or maybe we should wrap/modify the uf2conv script so that we can pass an optional offset base parameter?

Reply:
The `uf2conv.py` script is a direst import from https://github.com/microsoft/uf2 so I wouldn't want to just modify that w/o working w/ upstream.

It seems like that logic for the final offset is sane, but I'm also not sure you'd ever flash a UF2 without the 2nd stage bootloader, given that's usually loaded by MCUBoot in the discussed scenarios. Do we really need to solve for that for the UF2 output?